**В первом ящике находится 8 мячей, из которых 5 - белые. Во втором
ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают
случайным образом два мяча, из второго - 4. Какова вероятность того, что
3 мяча белые?**

In [8]:
import numpy as np
from math import factorial

Пусть $A$-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда
это событие может произойти следующими способами:  
а) $A_1$ из первой корзины не вытащили ни одного белого мяча, из второй
вытащили 3 белых мяча;  
б) $A_2$ из первой корзины вытащили 1 белый мяч, из второй вытащили 2
белых мяча;  
в) $A_3$ из первой корзины вытащили 2 белых мяча, из второй вытащили 1
белый мяч.  
Вероятность события $A$ будет определяться как сумма вероятностей этих
событий.  

In [9]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

Определим вероятность события $A_1$:  
$P(A_1)=\frac{C^2_3}{C^2_8}\cdot\frac{C^3_5\cdot C^1_7}{C^4_{12}}$

In [10]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(A1) = {P1: .4f}'
     )

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) =  0.0152


Определим вероятность события $A_2$:  
$P(A_2)=\frac{C^1_5\cdot C^1_3}{C^2_8}\cdot\frac{C^2_5\cdot C^2_7}{C^4_{12}}$

In [11]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(A2) = {P2: .4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) =  0.2273


Определим вероятность события $A_3$:  
$P(A_3)=\frac{C^2_5}{C^2_8}\cdot\frac{C^1_5\cdot C^3_7}{C^4_{12}}$

In [12]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(A3) = {P3: .4f}')

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A3) =  0.1263


Вычислим $P(A)=P(A_1+A_2+A_3)=P(A_1)+P(A_2)+P(A_3)$

In [13]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


------------------------------------------------------------------------

#### Ответ:

Вероятность того, что 3 мяча белые $P(A)=0.3687$ или $\approx 36.87\%$

------------------------------------------------------------------------